src/inspect_ai/model/_providers/goodfire.py

```diff
@@ -0,0 +1,244 @@
+from .._chat_message import (
+    ChatMessage,
+    ChatMessageAssistant,
+    ChatMessageSystem,
+    ChatMessageTool,
+    ChatMessageUser,
+)
+from .._generate_config import GenerateConfig
+from .._call_tools import Tool
+from .util import environment_prerequisite_error, model_base_url
+
+logger = logging.getLogger(__name__)
+
+# Constants
+GOODFIRE_API_KEY = "GOODFIRE_API_KEY"
+MIN_VERSION = "0.2.5"
+DEFAULT_BASE_URL = "https://api.goodfire.ai"
+DEFAULT_MAX_TOKENS = 4096
+DEFAULT_TEMPERATURE = 0.7
+DEFAULT_TOP_P = 0.95
+DEFAULT_MAX_CONNECTIONS = 10
+
+# Supported model mapping
+MODEL_MAP = {
+    "meta-llama/Meta-Llama-3-8B-Instruct": "meta-llama/Meta-Llama-3.1-8B-Instruct",
+    "meta-llama/Llama-3.3-70B-Instruct": "meta-llama/Llama-3.3-70B-Instruct",
+}
+
+class GoodfireAPI(ModelAPI):
+    """Goodfire API provider.
+    
+    This provider implements the Goodfire API for LLM inference. It supports:
+    - Chat completions with standard message formats
+    - Basic parameter controls (temperature, top_p, etc.)
+    - Usage statistics tracking
+    
+    Does not currently support:
+    - Tool calls
+    - Feature analysis
+    - Streaming responses
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        base_url: str | None = None,
+        api_key: str | None = None,
+        api_key_vars: list[str] = [],
+        config: GenerateConfig = GenerateConfig(),
+        **kwargs: Any,
+    ) -> None:
+        """Initialize the Goodfire API provider.
+        
+        Args:
+            model_name: Name of the model to use
+            base_url: Optional custom API base URL
+            api_key: Optional API key (will check env vars if not provided)
+            api_key_vars: Additional env vars to check for API key
+            config: Generation config options
+        """
+        super().__init__(
+            model_name=model_name,
+            base_url=base_url,
+            api_key=api_key,
+            api_key_vars=[GOODFIRE_API_KEY],
+            config=config,
+            **kwargs,
+        )
+
+        verify_required_version("Goodfire API", "goodfire", MIN_VERSION)
+
+        # Get API key from environment if not provided
+        if not self.api_key:
+            self.api_key = os.environ.get(GOODFIRE_API_KEY)
+        if not self.api_key:
+            raise environment_prerequisite_error("Goodfire", GOODFIRE_API_KEY)
+
+        # Format and validate model name
+        if not model_name.startswith("meta-llama/"):
+            model_name = f"meta-llama/{model_name}"
+
+        supported_models = list(get_args(SUPPORTED_MODELS))
+        if model_name not in supported_models:
+            raise ValueError(f"Model {model_name} not supported. Supported models: {supported_models}")
+
+        # Initialize client
+        base_url_val = model_base_url(base_url, "GOODFIRE_BASE_URL") 
+        assert isinstance(base_url_val, str) or base_url_val is None
+        self.client = goodfire.Client(
```

Comment:
As mentioned above, you likely want to spread `**model_args` here to allow users to specify any argument taken by the `goodfire.Client`.